T/179

Tade0 opened the pull request:
#179 Added CodePen and jsFiddle links to samples.

mlewand commented:
KK, please review this task.

f1ames commented:
Overall, it looks good, there are just few improvements to make:

1. While CodePen demo works fine, when opening JSFiddle demo it throws `Uncaught ReferenceError: CKEDITOR is not defined` and **Run** button needs to be clicked to reload the sample to make it work. It seems like some JSFiddle issue with async external js loading, not sure if we can do something about it?

2. In a few samples there are missing dependencies (like http://sdk.ckeditor.dev/offline/samples/bbcode.html missing _bbcode plugin_). I think we should link to `full-all` instead of `full` build of CKEditor.

3. Samples like http://sdk.ckeditor.dev/offline/samples/api.html has additional HTML code (in this case buttons to fire some events on the editor instance) and without it, it does not make much sense. If possible we should also append this Html in code samples. If not we should probably exclude such samples from creating links to JSFiddle/CodePen.
Also this sample has additional helper script (https://github.com/ckeditor/ckeditor-sdk/blob/master/samples/api.html#L40) which is not added to JSFiddle/CodePen code so the `Uncaught ReferenceError: ShowMessage is not defined` error is thrown.

4. From visual point of view JSFiddle and CodePen icons seems very small, we should probably make them bigger (maybe add some hover effect - but consistent with rest of sdk). See images below.

![sdk1](https://cloud.githubusercontent.com/assets/1061942/15532727/9ef466fe-2261-11e6-86e2-3700e13b4c3f.png)
![sdk2](https://cloud.githubusercontent.com/assets/1061942/15532726/9ef3917a-2261-11e6-86cf-2c5b576d4a11.png)

While we are using small images, there is no need to load it as 500x500px or 600x600px images, we should just use same size as we present on a page.

f1ames commented:
I noticed MDN (e.g. https://developer.mozilla.org/en-US/docs/Web/CSS/General_sibling_selectors) has the same links and it is nicely styled:
![mdn](https://cloud.githubusercontent.com/assets/1061942/16034519/943c2bc8-3212-11e6-8d27-4fe87f8d7364.png)

mlewand commented:
It does look very nice. As for "old/traditional" code listing popup, we could also put it inside a button like this. That was an issue for some time that it was looking like a standard link, a reference for related article.

Tade0 commented:
It seems that it's pretty easy to have that additional code that's in http://sdk.ckeditor.dev/offline/samples/api.html, but it still has some artifacts(mostly additional whitespace inside of textareas). I'm currently in the process of figuring out how to get rid of those.

Tade0 commented:
Update: I figured that instead of using variables that normally serve some arcane purpose I'll simply use the present, generated HTML output for the external code editors. Works a lot better than my former approach.

Also disabled samples that have `extraPlugins` somewhere in their scripts, because that's an indicator that the sample loads additional data, which obviously doesn't work in external editors.

Tade0 commented:
Summary of this ticket(based on https://github.com/ckeditor/ckeditor-sdk/pull/180#issuecomment-222621527):
1. I tried a few approaches, and a clean solution doesn't seem to exist. I guess it would be good to notify the users about this in the form of e.g. a comment in the script.
2. Using full-all now, and links to images are taken from the online version.
3. I managed to include this additional HTML in the sample and it works.
4. Images will be substituted with buttons in the future, so leaving it like this for now.

Also: I think that if we are to change the buttons, we need to do it consistently throughout the SDK. Perhaps an issue for this would be in order?

Comandeer commented:
1. I don't like the buttons… They definitely lack `[title]` attribute (not everyone knows all logos). The second issue with current buttons is the fact that they're totally inaccessible (there is no `[alt]` attribute for them!). But probably is not a big issue as we're going to switch to buttons.
2. Samples in http://sdk.ckeditor.dev/samples/fileupload.html are not working correctly – links to CKFinder are pointing to non-existent locations. Perhaps we could rewrite them in the same manner as other links to resources, but I'm not sure if we won't hit cross origin issues.
3. I'm really curious if there is nothing we can do about this issue with JSFiddle:

> While CodePen demo works fine, when opening JSFiddle demo it throws Uncaught ReferenceError: CKEDITOR is not defined and Run button needs to be clicked to reload the sample to make it work. 

Despite these little things, everything seems fine.

Tade0 commented:
That JSFiddle problem may have something to do with the fact, that the external resources are loaded with `http://...` instead of `https://...`. I'll change the protocol and we'll see if it works.

Tade0 commented:
1. Added title and alt.
2. I figured that since it's explicitly stated in the comment within sample: `Configure your file manager integration` the user is going understand, that this sample here can't work.
3. Switched to `https` (several people were complaining that external resources don't work if they're not served through `https`), but that didn't improve anything.
